addons/account/models/account_bank_statement_line.py

```diff
@@ -415,20 +415,22 @@ def create(self, vals_list):
             'name': False,
             **vals,
         } for vals in vals_list])
-
-        for i, (st_line, vals) in enumerate(zip(st_lines, vals_list)):
-            counterpart_account_id = counterpart_account_ids[i]
-
+        for st_line, vals in zip(st_lines, vals_list):
             to_write = {'statement_line_id': st_line.id, 'narration': st_line.narration, 'name': False}
-            if 'line_ids' not in vals_list[i]:
-                to_write['line_ids'] = [(0, 0, line_vals) for line_vals in st_line._prepare_move_line_default_vals(
-                    counterpart_account_id=counterpart_account_id)]
             with self.env.protecting(self.env['account.move']._get_protected_vals(vals, st_line)):
                 st_line.move_id.write(to_write)
-            self.env.add_to_compute(self.env['account.move']._fields['name'], st_line.move_id)
+        to_create_lines_vals = []
+        for i, (st_line, vals) in enumerate(zip(st_lines, vals_list)):
+            if 'line_ids' not in vals_list[i]:
+                to_create_lines_vals.extend(
+                    line_vals
+                    for line_vals in st_line._prepare_move_line_default_vals(counterpart_account_ids[i])
+                )
```

Comment:
Why making 2 loops instead of one?
```py
        to_create_lines_vals = []
        for i, (st_line, vals) in enumerate(zip(st_lines, vals_list)):
            to_write = {'statement_line_id': st_line.id, 'narration': st_line.narration, 'name': False}
            with self.env.protecting(self.env['account.move']._get_protected_vals(vals, st_line)):
                st_line.move_id.write(to_write)
            if 'line_ids' not in vals_list[i]:
                to_create_lines_vals.extend(
                    line_vals
                    for line_vals in st_line._prepare_move_line_default_vals(counterpart_account_ids[i])
                )
```